sdk/textanalytics/ai-text-analytics/CHANGELOG.md

```diff
@@ -1,6 +1,9 @@
 # Release History
 
-## 1.0.1 (Unreleased)
+## 1.1.0-preview.1 (unreleased)
+
+- Begin using version 3.1-preview.1 that has support for opinion mining, a new feature of Sentiment Analysis.
+- [Breaking] The `text` property of the `SentenceSentiment` interface is no longer optional because the service always returns it.
```

Comment:
This is not breaking the service (see my post on the teams channel) but rather the client. I would expect existing client code to guard against whether this property is undefined but now this logic is no longer required. I am not sure if this strictly falls within the breaking realm but was trying to be conservative.

Reply:
I see. The SDKs we ship have a guarantee that we won't introducing breaking changes after GA, so I think adding the "Breaking" might be alarming to some users, esp since I don't think it's actually breaking. I think in this case, we don't have to include it in the changelog since the behavior shouldn't change at all for the user

Reply:
I removed the breaking tag.

Reply:
The change would be client-breaking for Typescript users, maybe not for JS. For example in the below TS snippet, changing Foo's text property to required would cause this snippet to start failing on build

**Before:**

![image](https://user-images.githubusercontent.com/21109913/87817195-6443e780-c81d-11ea-8c09-65fad0e3aaab.png)
 
**Broken:**

![image](https://user-images.githubusercontent.com/21109913/87817328-9ce3c100-c81d-11ea-84a6-1b0f09babee5.png)

I think `SentenceSentiment` type would run into this.

With that said, it looks like it is a change we have to take. @ramya-rao-a any thoughts?

Reply:
@joheredi You're right this will fail. However, this is a response object and the user is not supposed to create it themselves.

Reply:
We should not make changes in the client library that would be considered breaking unless we make a major version bump of the library (for languages that support semver). And we only bump the major version when we absolutely positively have to. 

In this case, you can choose to continue to say that the property is optional and that is an extreme stroke of luck that every time you look, it is there. That would be much preferable over a new major version bump. 

From the service's perspective, this is not a breaking change.

Reply:
Oh! I didn't realize that this was part of the response. Yeah, I think that keeping `text` optional for this change makes total sense! And we can take the change the next time we need to release a major version.

Reply:
It may be of interest to talk to @bterlson about how we can best accommodate changes like this for output types. Making a field in an output should ideally not be breaking ever. And it could either (as proposed above) mean that our client libraries continue to keep the field as "optional" - or that we find some other mechanism by which we provide a fake default value for when someone tries to construct the item (most likely in a test case) without the property present....

Reply:
I intend for this change to be part of the version that will eventually supports opinion mining which I believe should be a major version bump, so keeping the PR as is should be fine I think. Is everyone ok with is?

Reply:
Why would the addition of a new feature (opinion mining) be targeting a new *major* version? It should be a minor version, no?

Reply:
I see.

Reply:
I talked to Brian and he considers the conversion from optional to required in an output type non-breaking, so I will keep the PR as is.

Reply:
If this type is output only, I would not consider this a breaking change. Pedantically someone could be using that type as an input type someplace but I don't worry too much about that without knowing a user scenario where someone would want to do it.

Reply:
For my edification; there is not a scenario in Typescript where one would do a check on an optional attribute that all of a sudden would become invalid/an error if the attribute is required? (For example what calling [`Nullable<T>.HasValue`](https://docs.microsoft.com/en-us/dotnet/api/system.nullable-1.hasvalue?view=netcore-3.1) would expose if an optional value was exposed as a `Nullable<int>` in C# and it moved to an `int` once it became required....

Reply:
@bterlson, the canonical scenario for creating output-types client side is mocking/testing.

Reply:
@johanste regarding your first comment, I could not come up with a similar example because I do not believe optional properties have extra properties/methods in them. Could you, @bterlson?
Regarding your second point, this is a good point. However, it comes down to whether we are supporting client code tests. Do we have established guidelines across all supported languages regarding what do we support exactly?

Reply:
I have been reading the docs and it looks like I should not mix patches and new features in the same version: https://azure.github.io/azure-sdk/typescript_implementation.html#ts-versioning
So I will close this PR for now and focus on the patch one and speak with the architecture board regarding this issue.